nix-buffer.el

```diff
@@ -216,9 +216,10 @@ is removed."
   (interactive)
   (let* ((root (directory-file-name (or (buffer-file-name) default-directory)))
 	 (expr-dir (locate-dominating-file root "dir-locals.nix")))
-    (when expr-dir
-	 (let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
-	   (nix-buffer--nix-build root expr-file)))))
+    (if expr-dir
+	(let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
+	  (nix-buffer--nix-build root expr-file))
+      (error "No dir-locals.nix file found"))))
```

Comment:
See https://github.com/NixOS/nixpkgs/blob/fd6e51d5505c128c2051ddcbcb395d12c0d94299/pkgs/applications/science/logic/coq/default.nix#L60-L68 for what I had to do to get `ProofGeneral`'s use of `scomint` working.